### PR TITLE
Registry config javadoc

### PR DESCRIPTION
**Pull request by aloubyansky**
Added missing javadoc.

### CODE REVIEW

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/config/RegistryPlatformsConfig.java [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
 
+    /**
+     * Whether the client should send requests to resolve the platform extension catalogs (platform descriptors)
+     * to the registry or resolve them from Maven Central directly instead.
+     * Returning <code>null</code> from this method will be equivalent to returning <code>false</code>, in which case
+     * the client will not send requests to resolve platform extension catalogs to the registry.
+     *
+     * @return true if the registry will be able to handle platform descriptor requests, otherwise - false
+     */
     Boolean getExtensionCatalogsIncluded();
```
Comment (gastaldi) [inline independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/config/RegistryPlatformsConfig.java:16]: Why this method is not returning a primitive?

**Review by gastaldi — APPROVED** [review]
LGTM, added one comment

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/config/RegistryPlatformsConfig.java (hunk 1 above)
Comment (aloubyansky) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/config/RegistryPlatformsConfig.java:16]: The way it is implemented currently, the default config is provided by the registry itself. The client may have a customized version of it (i.e. by selectively overriding some values) in the `config.yaml`. When the registry client is initialized, it is merging the default registry config and the client side version (if it exists). A non-null value from this method would represent a user's choice, which will override the registry's default. So `null` in this case will mean that the user is willing to accept whatever the registry's default is.

Comment (gastaldi) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/config/RegistryPlatformsConfig.java:16]: I'd recommend to put that in the Javadoc, but too late now :)

**Pull request merged**